pkg/tests/cross-tests/tf_driver.go

```diff
@@ -120,6 +120,9 @@ func (d *tfDriver) coalesce(t T, x any) *tftypes.Value {
 		return nil
 	}
 	objectType := convert.InferObjectType(sdkv2.NewSchemaMap(d.res.Schema), nil)
+	for k := range objectType.AttributeTypes {
+		objectType.OptionalAttributes[k] = struct{}{}
```

Comment:
To be able to express missing values, rather than nils, in cross-tests, just assume all attributes are optional. I think this should work out.